api/pkg/service/catalog/syncer.go

```diff
@@ -405,3 +429,39 @@ func (s *syncer) updateCatalogResult(
 	}
 	return nil
 }
+
+// Delete the resource by id
+func (s *syncer) deleteResource(txn *gorm.DB, log *zap.SugaredLogger, res model.Resource) {
+
+	txn.Unscoped().Where("id = ?", res.ID).Delete(&model.Resource{})
+
+	log.Infof("Resource %s of kind %s has been deleted", res.Name, res.Kind)
+}
+
+// Delete the tags which doesn't belongs to any other existing resources
+func (s *syncer) deleteTag(txn *gorm.DB, log *zap.SugaredLogger, tags []model.ResourceTag) {
```

Comment:
do we need to do this as part of this api ? 🤔  what if we have a clean up api which would remove things like this which are no longer required

Reply:
I think yes , if we want to have a clean up api then how we would identify which things are no longer required ?